### ISSUE TITLE
tempnam() error, while using hostings :)

### ISSUE THREAD

**Issue by Kharalampidi** [thephpleague/glide#186]
After uploading application on hosting I have some errors. Because '/tmp' dir (from sys_get_temp_dir()) is not accessable for my user. Can you check 'upload_tmp_dir' php.ini configuration?
https://github.com/thephpleague/glide/blob/master/src/Server.php#L515
Maybe:
```
        $tmpDir = !empty(ini_get('upload_tmp_dir')) ? ini_get('upload_tmp_dir') : sys_get_temp_dir();
        $tmp = tempnam($tmpDir, 'Glide');
```

Thank you!

**Comment by arnaupc** [comment]
I have the same issue in a shared hosting and found the same solution.
It will be great if you merge into master.
Thanks!

**Comment by wbadrh** [comment]
Concept idea:

Condition | Use
--- | ---
Default | `sys_get_temp_dir()`
No write access | `upload_tmp_dir` (php.ini)
No write access (or empty) | `$cache_folder`
$cache_folder != local adapter | Throw exception

```
$cache_folder = "path/to/cache/folder";
$cache_folder = new MemoryAdapter(); // won't work
```
```
$server = League\Glide\ServerFactory::create([
    "source" => "path/to/source/folder",
    "cache" => $cache_folder
]);

...
```


```
$tmp = sys_get_temp_dir();

if (!is_writable($tmp)) {
  $tmp = ini_get("upload_tmp_dir");

  if (!is_writable($tmp)) {
    if ($local_adapter) {
      $tmp = $cache_folder;
    } else {
      throw new FilesystemException("No write access to temporary folder.");
    }
  }
}
```

**Comment by jaapio** [comment]
I think I solved this in #165

**Comment by tgalopin** [comment]
Should be fixed in master, I will release in the coming days.
